### PR TITLE
Make it possible to reattach NEGs automatically

### PR DESCRIPTION
**Pull request by fdfzcq**
The current behaviour of Autoneg introduces race conditions for us, the scenario goes:

- A service is deployed with existing `BackendService` and NEGs
- Autoneg connects the service to existing NEGs and updates its annotations
- User removes `BackendService` resource
- Autoneg does nothing because the NEGs status stays unchanged but the service is actually detached from the NEGs
- User recreates a new `BackendService`
- Autoneg will do nothing because the NEGs status stays unchanged

The proposed solution here without us doing big architectural change is to make Autoneg reconcile periodically.
Since this is a big behaviour change, I added a flag to enable it but defaults to false. Along with another flag to configure the reconciliation period.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,6 +82,10 @@ The controller parameters can be customized via changing the [controller deploym
   to false. The template defaults to `{name}-{port}`. It can contain `namespace`, `name`, `port` and `hash` and the non-hash values will be 
   truncated evenly if the full name is longer than 63 characters. `<hash>` is generated using full length `namespace`, `name` and 
   `port` to avoid name collisions when truncated.
+* `--max-rate-per-endpoint`: optional. Sets a default value for max-rate-per-endpoint that can be overriden by user config. Defaults to 0.
```
Comment (rosmo) [inline README.md:85]: Also typo: `overriden`

**Comment by rosmo** [comment]
Don't get me wrong, I support this, but couldn't the user just delete the status annotation to re-reconcile?

**Comment by fdfzcq** [comment]
@rosmo Yes, that was a workaround, but it adds toils. And we think this kind of mishaps can happen without users notice and cause incident.

**Comment by rosmo** [comment]
LGTM

**Comment by joshw123** [comment]
Hello, Do you know when this will be released? :) Thank You!

**Comment by rosmo** [comment]
@joshw123 I'll roll out a new pre-release after issue in #113 is solved.

**Pull request merged**